app/controllers/api/users.php

```diff
@@ -97,7 +107,18 @@ function createUser(string $hash, mixed $hashOptions, string $userId, ?string $e
             }
         }
 
-        $password = (!empty($password)) ? ($hash === 'plaintext' ? Auth::passwordHash($password, $hash, $hashOptionsObject) : $password) : null;
+        $hashedPassword = null;
+
+        if (!empty($password)) {
+            if ($hash instanceof Plaintext) { // Password was never hashed, hash it with the default hash
+                $defaultHash = new ProofsPassword();
```

Comment:
Can we inject the resource?

Reply:
I think it's better this way? only one endpoint sends plaintext to this function, and if we inject, we will have to inject in all the endpoints to create user using various different hash.